backend/src/backend_api.cpp

```diff
@@ -5,8 +5,8 @@
 void runBackend(const int mavlink_listen_port, void (*onServerStarted)(void *), void *context)
 {
     dronecode_sdk::backend::DronecodeSDKBackend backend;
-    backend.connect(mavlink_listen_port);
     backend.startGRPCServer();
+    backend.connect(mavlink_listen_port);
```

Comment:
Hmm, then that could be done after the callback (line 11) is called, I believe.

As discussed, this change makes the callback superfluous. But because `backend.connect()` is waiting for the discovery, calling `backend.connect()` before or after the callback (line 11) will change the behavior on the user side. So let's try to first call the callback, and then wait for the connection, so that the callback is truly superfluous :D.

Not sure if that makes sense at all :sweat_smile:.

Reply:
But it works? So what should I change?

Reply:
Well nevermind, maybe that's actually clearer like this, because on the client side we call it `Core.connect()` and not `Core.start()`.